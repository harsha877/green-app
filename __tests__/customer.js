import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import Customer from '../src/views/customer';

test("Customer Screen Loading",()=>{
    const push = jest.fn();
   render(<Customer navigation={{ push }} />)

})

test("Customer Screen snapshot",()=>{
    const push = jest.fn();
   let CustomerComponent=render(<Customer navigation={{ push }} />).toJSON();
   expect(CustomerComponent).toMatchSnapshot();
})

describe("Customer Screen QueryTesting",()=>{
    const push = jest.fn();
    let CustomerComponent;
    beforeEach(()=>{
    CustomerComponent=render(<Customer navigation={{ push }}/>);
    })

    test("Testing Styles",()=>{
        const props=CustomerComponent.toJSON();
        let containerStyles=props.props.style;
        expect(containerStyles).toStrictEqual({
            display: "flex",
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
            marginTop: 30
        });})

});