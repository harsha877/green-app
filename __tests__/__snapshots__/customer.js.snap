// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Screen snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": 30,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "#2471A3",
            "fontSize": 30,
            "fontWeight": "500",
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        Customer Login
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 100,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/avatar.png",
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="username"
          style={
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 40,
              "margin": 15,
              "padding": 10,
              "width": 300,
            }
          }
          value=""
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 40,
              "margin": 15,
              "padding": 10,
              "width": 300,
            }
          }
          value=""
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              LOGIN
            </Text>
          </View>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "#2471A3",
              "height": 80,
              "marginTop": 250,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={200}
            bbWidth={750}
            focusable={false}
            height={200}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 200,
                  "width": 750,
                },
              ]
            }
            vbHeight={320}
            vbWidth={1440}
            width={750}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M0,192L60,170.7C120,149,240,107,360,112C480,117,600,171,720,197.3C840,224,960,224,1080,208C1200,192,1320,160,1380,144L1440,128L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
                fill={4280578467}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "bottom": 10,
            "position": "absolute",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2471A3",
              "height": 80,
              "marginTop": 250,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={200}
            bbWidth={750}
            focusable={false}
            height={200}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "bottom": 20,
                  "position": "absolute",
                },
                Object {
                  "flex": 0,
                  "height": 200,
                  "width": 750,
                },
              ]
            }
            vbHeight={320}
            vbWidth={1440}
            width={750}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M0,64L40,96C80,128,160,192,240,202.7C320,213,400,171,480,149.3C560,128,640,128,720,154.7C800,181,880,235,960,218.7C1040,203,1120,117,1200,74.7C1280,32,1360,32,1400,32L1440,32L1440,320L1400,320C1360,320,1280,320,1200,320C1120,320,1040,320,960,320C880,320,800,320,720,320C640,320,560,320,480,320C400,320,320,320,240,320C160,320,80,320,40,320L0,320Z"
                fill={4280578467}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
